templates/c/src/tic80.h

```diff
@@ -0,0 +1,262 @@
+#define MAP                ((uint8_t*) 0x8000)
+#define GAMEPADS           ((uint8_t*) 0xFF80)
+#define MOUSE              ((uint8_t*) 0xFF84)
+#define KEYBOARD           ((uint8_t*) 0xFF88)
+#define SFX_STATE          ((uint8_t*) 0xFF8C)
+#define SOUND_REGISTERS    ((uint8_t*) 0xFF9C)
+#define WAVEFORMS          ((uint8_t*) 0xFFE4)
+#define SFX                ((uint8_t*) 0x100E4)
+#define MUSIC_PATTERNS     ((uint8_t*) 0x11164)
+#define MUSIC_TRACKS       ((uint8_t*) 0x13E64)
+#define SOUND_STATE        ((uint8_t*) 0x13FFC)
+#define STEREO_VOLUME      ((uint8_t*) 0x14000)
+#define PERSISTENT_RAM     ((uint8_t*) 0x14004)
+#define PERSISTENT_RAM_u32 ((uint8_t*) 0x14004)
+#define SPRITE_FLAGS       ((uint8_t*) 0x14404)
+#define SYSTEM_FONT        ((uint8_t*) 0x14604)
+
+
+#define PALETTE            ((uint8_t*) 0x3FC0)
+#define PALETTE_MAP        ((uint8_t*) 0x3FF0)
+#define BORDER_COLOR       ((uint8_t*) 0x3FF8)
+#define SCREEN_OFFSET      ((uint8_t*) 0x3FF9)
+#define MOUSE_CURSOR       ((uint8_t*) 0x3FFB)
+#define BLIT_SEGMENT       ((uint8_t*) 0x3FFC)
+
+// vbank 1
+#define OVR_TRANSPARENCY   ((uint8_t*) 0x3FF8)
+
+
+
+// ┌───────────────────────────────────────────────────────────────────────────┐
+// │                                                                           │
+// │ API                                                                       │
+// │                                                                           │
+// └───────────────────────────────────────────────────────────────────────────┘
+
+// INPUT
+
+typedef struct {
+  int16_t x;
+  int16_t y;
+  int8_t scrollx;
+  int8_t scrolly;
+  bool left;
+  bool middle;
+  bool right;
+} MouseData;
+
+WASM_IMPORT("btn")
+uint32_t btn(int32_t id);
+
+WASM_IMPORT("btnp")
+uint32_t btnp(int32_t id , int32_t hold, int32_t period);
+
+WASM_IMPORT("key")
+bool key(int32_t keycode);
+
+WASM_IMPORT("keyp")
+bool keyp(int32_t keycode, int32_t hold, int32_t period);
+
+WASM_IMPORT("mouse")
+void mouse(MouseData* data);
+
+
+// DRAW / DRAW UTILS
+
+WASM_IMPORT("clip")
+void clip(int32_t x, int32_t y, int32_t w, int32_t h);
+
+WASM_IMPORT("cls")
+void cls(uint8_t color);
+
+WASM_IMPORT("circ")
+void circ(int32_t x, int32_t y, int32_t radius, uint8_t color);
+
+WASM_IMPORT("circb")
+void circb(int32_t x, int32_t y, int32_t radius, uint8_t color);
+
+WASM_IMPORT("elli")
+void elli(int32_t x, int32_t y, int32_t a, int32_t b, uint8_t color);
+
+WASM_IMPORT("ellib")
+void ellib(int32_t x, int32_t y, int32_t a, int32_t b,  uint8_t color);
+
+WASM_IMPORT("fget")
+bool fget(int32_t id, uint8_t flag);
+
+WASM_IMPORT("fset")
+void fset(int32_t id, uint8_t flag, bool value);
+
+WASM_IMPORT("line")
+void line(int32_t x0, int32_t y0, int32_t x1, int32_t y1, uint8_t color);
+
+WASM_IMPORT("map")
+void map(int32_t x, int32_t y, int32_t w, int32_t h, int32_t sx, int32_t sy,
+	  uint8_t* trans_colors, uint8_t colorCount,  int32_t scale,  int32_t remap);
+// todo: why is remap a uint32_t?
```

Comment:
It should be a  function pointer, but we don't know exactly what signature yet... a u32 is just a placeholder... function pointers are also u32, so changing this later shouldn't beak anything.